neurom/ezy/neuron.py

```diff
@@ -110,6 +112,12 @@ def __init__(self, neuron, iterable_type=np.array):
         super(Neuron, self).__init__(neuron.soma, neuron.neurites, neuron.name)
         self._iterable_type = iterable_type
 
+    def __eq__(self, other):
+        return False if not isinstance(self, type(other)) else \
+               all(self._compare_neurites(other, ttype) for ttype in
```

Comment:
Because this is faster due to the fact that if I compare the neurites of one neuron and the neurites of another neuron then I will have (NBASAL + NAPICAL + NUNDEFINED + 1)^2 possible combinations (I do not assume that the order the neurites are stored is sorted), while if I separate them by time I will not have for example to compare the axon with the basals etc. having thus NBASAL^2 + NAPICAL^2 + NUNDEFINED^2 + 1 combinations. I hope I did the math correctly :P